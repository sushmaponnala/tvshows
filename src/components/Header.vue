<template>
    <div>
        <b-navbar toggleable="lg" type="dark" variant="dark">
            <router-link class="navbar-brand" to="/">TV Shows</router-link>
            <b-navbar-toggle target="nav-collapse"></b-navbar-toggle>
            <b-collapse id="nav-collapse" is-nav>
                <b-navbar-nav>
                    <router-link class="nav-link" to="/">Home</router-link>
                </b-navbar-nav>

                <b-navbar-nav class="ml-auto">
                    <b-nav-form v-on:submit.prevent="onSubmit">
                        <b-form-input type="search" size="sm" class="mr-sm-2" v-model="query" placeholder="Search"></b-form-input>
                        <b-button size="sm" class="my-2 my-sm-0" type="submit">Go</b-button>
                    </b-nav-form>
                </b-navbar-nav>
            </b-collapse>
        </b-navbar>
    </div>
</template>

<script>
export default {
  name: 'Header',
  data(){
    return {
        query: ""
    }
  },
  methods: {
      onSubmit() {
          if(this.query){
            this.$router.push({ path: `/search?q=${this.query}` });
            this.query = ""; // reset search field
          }
      }
  }
}
</script>