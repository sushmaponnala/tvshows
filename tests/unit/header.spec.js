import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import Header from '@/components/Header.vue'
import { BootstrapVue } from 'bootstrap-vue'

const localVue = createLocalVue()

localVue.use(BootstrapVue)
localVue.use(VueRouter)

describe('Header.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount(Header, {
            propsData: {},
            localVue
        })
        expect(wrapper.element).toMatchSnapshot();
    })
})
