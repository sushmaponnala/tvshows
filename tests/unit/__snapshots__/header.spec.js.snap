// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue renders correctly 1`] = `
<div>
  <b-navbar-stub
    tag="nav"
    toggleable="lg"
    type="dark"
    variant="dark"
  >
    <router-link-stub
      ariacurrentvalue="page"
      class="navbar-brand"
      event="click"
      tag="a"
      to="/"
    >
      TV Shows
    </router-link-stub>
     
    <b-navbar-toggle-stub
      label="Toggle navigation"
      target="nav-collapse"
    />
     
    <b-collapse-stub
      id="nav-collapse"
      isnav="true"
      tag="div"
    >
      <b-navbar-nav-stub
        tag="ul"
      >
        <router-link-stub
          ariacurrentvalue="page"
          class="nav-link"
          event="click"
          tag="a"
          to="/"
        >
          Home
        </router-link-stub>
      </b-navbar-nav-stub>
       
      <b-navbar-nav-stub
        class="ml-auto"
        tag="ul"
      >
        <b-nav-form-stub>
          <b-form-input-stub
            class="mr-sm-2"
            debounce="0"
            placeholder="Search"
            size="sm"
            type="search"
            value=""
          />
           
          <b-button-stub
            class="my-2 my-sm-0"
            size="sm"
            tag="button"
            target="_self"
            type="submit"
            variant="secondary"
          >
            Go
          </b-button-stub>
        </b-nav-form-stub>
      </b-navbar-nav-stub>
    </b-collapse-stub>
  </b-navbar-stub>
</div>
`;
